src/runtime/heap/objcache.c

```diff
@@ -20,27 +20,28 @@
 
 #include <runtime.h>
 
-#define FOOTER_MAGIC	(u16)(0xcafe)
+#define FOOTER_MAGIC    (u16)(0xcafe)
 
 typedef struct objcache *objcache;
 typedef struct footer {
-    u16 magic;			/* try to detect corruption by overruns */
-    u16 free;			/* next free (recycled) object in page */
-    u16 head;			/* next uninitialized object in page */
-    u16 avail;			/* # of free and uninit. objects in page */
-    objcache cache;		/* objcache to which this page belongs */
-    struct list list;		/* full list if avail == 0, free otherwise */
+    u16 magic;              /* try to detect corruption by overruns */
+    u16 free;               /* next free (recycled) object in page */
+    u16 head;               /* next uninitialized object in page */
+    u16 avail;              /* # of free and uninit. objects in page */
+    heap cache;             /* objcache to which this page belongs */
+    struct list list;       /* full list if avail == 0, free otherwise */
 } *footer;
 
 typedef struct objcache {
     struct heap h;
     heap parent;
-    struct list free;		/* pages with available objects */
-    struct list full;		/* fully-occupied pages */
-    bytes pagesize;		/* allocation size for parent heap */
-    u64 objs_per_page;		/* objects per page */
-    u64 total_objs;		/* total objects in cache */
-    u64 alloced_objs;		/* total cache occupancy (of total_objs) */
+    struct list free;       /* pages with available objects */
+    struct list full;       /* fully-occupied pages */
+    bytes pagesize;         /* allocation size for parent heap */
+    u64 objs_per_page;      /* objects per page */
+    u64 total_objs;         /* total objects in cache */
+    u64 alloced_objs;       /* total cache occupancy (of total_objs) */
+    heap wrapper_heap;      /* heap wrapper */
```

Comment:
This new struct member should be initialized in allocate_objcache(), otherwise it could get a random value when a (non-debug) objcache is created.

Reply:
Yup, properly initialized now!